cmd/auth-rpc-client.go

```diff
@@ -127,30 +128,50 @@ func (authClient *AuthRPCClient) Close() error {
 }
 
 // Login - a jwt based authentication is performed with rpc server.
-func (authClient *AuthRPCClient) Login() error {
+func (authClient *AuthRPCClient) Login() (err error) {
 	authClient.mu.Lock()
 	defer authClient.mu.Unlock()
+
+	// As soon as the function returns,  update error count
+	// if there is an error.
+	defer func() {
+		if err != nil {
+			authClient.serverIOErrCnt++
+		}
+	}()
+
+	// Take remote disk offline if the total server errors
+	// are more than maximum allowable IO error limit.
+	if authClient.serverIOErrCnt > maxAllowedIOError {
```

Comment:
Better to put this `if` block before calling the defer. If the threshold errors have been reached, then calling the defer it unnecessary.